heka/sandbox/encoders/combine_telemetry_objects.lua

```diff
@@ -0,0 +1,34 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+require "cjson"
+local l = require "lpeg"
+
+local grammar = (l.C"payload" + l.C"environment") * l.P"." * l.C(l.P(1)^1)
+
+function process_message()
+    local raw = read_message("raw")
+    local ok, msg = pcall(decode_message, raw)
+    if not ok then return -1, msg end
+
+    if type(msg.Fields) ~= "table" then return -1, "missing Fields" end
+
+    local ok, json = pcall(cjson.decode, read_message("Payload"))
+    if not ok then return -1, json end
+
+    for i=1, #msg.Fields do
+        local section, name = grammar:match(msg.Fields[i].name)
+        if section then
+            local ok, object = pcall(cjson.decode, msg.Fields[i].value[1])
+            if ok then
+                json[section][name] = object
+            end
+        end
+    end
+    local ok, payload = pcall(cjson.encode, json)
+    if not ok then return -1, payload end
+
+    inject_payload("txt", "output", msg.clientId, "\t", payload, "\n")
```

Comment:
The previous data export included the leftover Fields in the output - there are a few things that are not inside the payload that could be useful, though I'm not sure whether they are neccessary or not.